### PR TITLE
rclcpp logging still uses fprintf all over the place.

### PR DESCRIPTION
**Pull request by guillaumeautran**
Remove all printf log lines and replace with RCLUTILS_LOG_XXX macros.

Issue: #438

### CODE REVIEW

File: rclcpp/src/rclcpp/utilities.cpp [hunk 1]
```diff
@@ -114,8 +117,9 @@ trigger_interrupt_guard_condition(int signal_value)
     for (auto & kv : g_sigint_guard_cond_handles) {
       rcl_ret_t status = rcl_trigger_guard_condition(&(kv.second));
       if (status != RCL_RET_OK) {
-        fprintf(stderr,
-          "[rclcpp::error] failed to trigger guard condition: %s\n", rcl_get_error_string_safe());
+        RCUTILS_LOG_ERROR_NAMED(
+        "rclcpp",
```
Comment (mikaelarguedas) [inline rclcpp/src/rclcpp/utilities.cpp]: Nit: this should be indented

File: rclcpp/include/rclcpp/client.hpp [hunk 2]
```diff
@@ -39,6 +39,9 @@
 #include "rmw/error_handling.h"
 #include "rmw/rmw.h"
 
+#include "rcutils/logging_macros.h"
```
Comment (mikaelarguedas) [inline rclcpp/include/rclcpp/client.hpp]: Nit: we encourage the use of single vertical space

**Comment by guillaumeautran** [comment]
@mikaelarguedas I'm having problem running the tests on my system due to some gmock cmake non-sense.  Any idea?

```
-- Found gmock sources under '/usr/src/gmock': C++ tests using 'Google Mock' will be built
CMake Error at /usr/src/gtest/cmake/internal_utils.cmake:130 (add_library):
  add_library cannot create target "gtest" because another target with the
  same name already exists.  The existing target is a static library created
  in source directory "/usr/src/gtest".  See documentation for policy CMP0002
  for more details.
Call Stack (most recent call first):
  /usr/src/gtest/cmake/internal_utils.cmake:153 (cxx_library_with_type)
  /usr/src/gtest/CMakeLists.txt:70 (cxx_library)


CMake Error at /usr/src/gtest/cmake/internal_utils.cmake:130 (add_library):
  add_library cannot create target "gtest_main" because another target with
  the same name already exists.  The existing target is a static library
  created in source directory "/usr/src/gtest".  See documentation for policy
  CMP0002 for more details.
Call Stack (most recent call first):
  /usr/src/gtest/cmake/internal_utils.cmake:153 (cxx_library_with_type)
  /usr/src/gtest/CMakeLists.txt:71 (cxx_library)


CMake Error at /usr/src/gtest/CMakeLists.txt:72 (target_link_libraries):
  Attempt to add link library "gtest" to target "gtest_main" which is not
  built in this directory.
```

File: rclcpp/include/rclcpp/client.hpp (hunk 2 above)
Comment (guillaumeautran) [reply rclcpp/include/rclcpp/client.hpp]: fixed

File: rclcpp/src/rclcpp/executor.cpp [hunk 3]
```diff
@@ -28,6 +28,9 @@
 
 #include "rcl_interfaces/msg/intra_process_message.hpp"
 
+#include "rcutils/logging_macros.h"
+
+
```
[the remark this answers is not in the document]
Comment (guillaumeautran) [reply rclcpp/src/rclcpp/executor.cpp]: fixed

File: rclcpp/src/rclcpp/publisher.cpp [hunk 4]
```diff
@@ -33,6 +33,9 @@
 #include "rclcpp/node.hpp"
 #include "rclcpp/expand_topic_or_service_name.hpp"
 
+#include "rcutils/logging_macros.h"
+
+
```
[the remark this answers is not in the document]
Comment (guillaumeautran) [reply rclcpp/src/rclcpp/publisher.cpp]: fixed

File: rclcpp/src/rclcpp/utilities.cpp [hunk 5]
```diff
@@ -26,10 +26,13 @@
 
 #include "rcl/error_handling.h"
 #include "rcl/rcl.h"
+#include "rcutils/logging_macros.h"
 
 #include "rmw/error_handling.h"
 #include "rmw/rmw.h"
 
+#include "rcutils/logging_macros.h"
```
[the remark this answers is not in the document]
Comment (guillaumeautran) [reply rclcpp/src/rclcpp/utilities.cpp]: fixed

File: rclcpp/src/rclcpp/utilities.cpp (hunk 1 above)
Comment (guillaumeautran) [reply rclcpp/src/rclcpp/utilities.cpp]: fixed

**Comment by wjwwood** [comment]
@guillaumeautran please include more context for that error, ideally all of the build output in a gist.github.com or something.

**Comment by guillaumeautran** [comment]
Ok, I sorted out my issue. Caused by: https://cmake.org/cmake/help/v3.0/policy/CMP0002.html since my cmake version is 3.5.1
But I took care of it and am now able to address the review comments. Patch coming up soon.

**Comment by dhood** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3977/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1090/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3314/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4079/)

**Comment by dhood** [comment]
This looks good thanks @guillaumeautran 

I was going to suggest the use of `ROS_PACKAGE_NAME` for the logger name but it's provided by `ament_cmake_ros` which isn't be used by this package yet (https://github.com/ros2/rclcpp/pull/444)

**Pull request merged**